backend/supabase/mod_reports.sql

```diff
@@ -0,0 +1,33 @@
+create type status_type as enum('new', 'under review', 'resolved', 'needs admin');
+
+create table if not exists
+  mod_reports (
+    report_id SERIAL primary key,
+    created_time timestamptz not null default now(),
+    comment_id text not null,
+    contract_id text not null,
+    user_id text not null,
+    status status_type not null default 'new',
+    contract_slug text not null,
+    contract_question text not null,
+    content jsonb not null,
+    creator_username text not null,
```

Comment:
I understand what you want me to do and the reasoning for it.

Struggling with the execution. GPT first suggested using .leftjoin which didn't work so then started using raw SQL. But there was a lot of errors and the approach started requiring a lot of new lines of code including creating a new custom postgress function. Maybe this is correct but I wanted to check if there is a method you suggest to join the queries.

Also:
user_id
comment_id
contract_id
created_time

are all duplicated from contract_comments. Should I delete all of these too? Intuitively I would guess it is fine to delete all of them except comment_id because it would still be good to keep one identifier for which comment as a property on the report itself?

Reply:
ID's are more often fine to copy around, but you could remove them except comment_id if you want. I'd probably keep them for convenience.

I think it's nice to have a created_time for this specific row in mod_reports, so I would leave that.

How do you join in the endpoint to get the comment data? Something like this:

```
const fullModReports = await pg.many(`
  select mr.*, cc.content, contracts.question, contracts.slug, users.username
  from mod_reports mr
  join contract_comments cc on cc.comment_id = mr.comment_id
  join contracts on contracts.id = mr.contract_id
  join users on users.id = contracts.creator_id
`)
```